kythe/go/platform/analysis/local/local.go

```diff
@@ -140,3 +145,103 @@ type kzipFetcher struct{ r *kzip.Reader }
 
 // Fetch implements the required method of analysis.Fetcher.
 func (k kzipFetcher) Fetch(_, digest string) ([]byte, error) { return k.r.ReadAll(digest) }
+
+// localAnalyzer provides an implementation of the analysis.CompilationAnalyzer
+// interface that works by saving the compilation unit to a temporary kzip file
+// and invoking an indexer binary on it.
+type localAnalyzer struct {
+	cmd     []string
+	fetcher analysis.Fetcher
+}
+
+// NewLocalAnalyzer creates an analyzer that will execute the given command when
+// Analyze() is called. The filepath to a kzip is appended to the command, so if
+// given ["my_indexer", "--index"], the final executed command will be
+// ["my_indexer", "--index", "some/file.kzip"]. The indexer command should emit
+// length-delimited Entry protos to stdout as described in
+// //kythe/go/platform/delimited.
+func NewLocalAnalyzer(cmd []string, fetcher analysis.Fetcher) (*localAnalyzer, error) {
```

Comment:
A couple things:

1. This name stutters with the package name. Consider naming the type `Analyzer` and the constructor `NewAnalyzer`, so that at the usage site the reader sees `local.NewAnalyzer` and the type `*local.Analyzer`.

2. Although it is legal to do so, I would discourage you from having an exported function or method return an unexported type. Better, I think to export the type (though not its fields). 

Other comments: As an API design issue, I would probably not bother having this constructor return an error at all—and either document that `cmd` is required and panic, or define it so that a zero-valued Analyzer is ready for use (e.g., by having an empty command do nothing). Panicking should, of course, be used with care—but in this case a program that passes an empty command or a nil fetcher is already statically wrong—i.e., it's a programming error rather than a runtime property. A more expressive type system might prevent it from occurring ("`[n≥1]string` type anyone?), but in lieu of that a panic is OK.

That said: The main things I suggest fixing here are the names and the exported-ness. 🙂

Reply:
Re: package names & stutter, cf. https://golang.org/doc/effective_go.html#package-names

Reply:
:+1: